src/System.Private.ServiceModel/tools/test/SelfHostWcfService/TestResources/HttpsResource.cs

```diff
@@ -19,20 +21,10 @@ protected override int GetPort(ResourceRequestContext context)
 
         protected override void ModifyHost(ServiceHost serviceHost, ResourceRequestContext context)
         {
-            EnsureHttpsCertificateInstalled(context.BridgeConfiguration);
-            base.ModifyHost(serviceHost, context);
-        }
-
-        internal static string EnsureHttpsCertificateInstalled(BridgeConfiguration configuration)
-        {
             // Ensure the https certificate is installed before this endpoint resource is used
-            string certThumbPrint = CertificateManager.InstallMyCertificate(configuration,
-                                                                            configuration.BridgeHttpsCertificate);
-
-            // Ensure http.sys has been told to use this certificate on the https port
-            CertificateManager.InstallSSLPortCertificate(certThumbPrint, configuration.BridgeHttpsPort);
+            CertificateResourceHelpers.EnsureSslPortCertificateInstalled(context.BridgeConfiguration);
```

Comment:
Not now, but this call to ensure the SSL port cert is installed feels like it should be the responsibility of one of the certificate resources.  It will be an easy thing to mess up adding new endpoint resources.